src/others/xml/parsers/lv_xml_obj_parser.c

```diff
@@ -309,18 +326,21 @@ void lv_obj_xml_subject_set_apply(lv_xml_parser_state_t * state, const char ** a
         return;
     }
 
-    bool type_ok = (subject->type == LV_SUBJECT_TYPE_INT && int_subject) ||
-                   (subject->type == LV_SUBJECT_TYPE_STRING && !int_subject);
-    if(!type_ok) {
+    if(subject->type != subject_type) {
         LV_LOG_WARN("`%s` subject has incorrect type in <lv_obj-subject_set>", subject_str);
         return;
     }
 
     void * item = lv_xml_state_get_item(state);
-    if(int_subject) {
+    if(subject_type == LV_SUBJECT_TYPE_INT) {
         lv_obj_add_subject_set_int_event(item, subject, trigger, lv_xml_atoi(value_str));
     }
-    else {
+#if LV_USE_FLOAT
+    else if(subject_type == LV_SUBJECT_TYPE_FLOAT) {
+        lv_obj_add_subject_set_float_event(item, subject, trigger, lv_xml_atof(value_str));
+    }
+#endif
```

Comment:
Here and in other places in this PR that there's a similar check, I believe it would make it more debug friendly to have the `else if` all the time and log an error if `LV_USE_FLOAT` is not set, basically this:

```suggestion
    else if(subject_type == LV_SUBJECT_TYPE_FLOAT) {
#if LV_USE_FLOAT
        lv_obj_add_subject_set_float_event(item, subject, trigger, lv_xml_atof(value_str));
#else
        LV_LOG_ERROR("Tried to add a subject of type float but LV_USE_FLOAT is not enabled");
#endif
    }
```